src/transaction/log_manager.c

```diff
@@ -2748,22 +2747,16 @@ log_append_postpone (THREAD_ENTRY * thread_p, LOG_RCVINDEX rcvindex, LOG_DATA_AD
     {
       return;
     }
-  if (VACUUM_IS_THREAD_VACUUM (thread_p))
-    {
-      /* Cache postpone log record. Redo data must be saved before calling prior_lsa_next_record, which may free this
-       * prior node. */
-      vacuum_cache_log_postpone_redo_data (thread_p, node->data_header, node->rdata, node->rlength);
-      // todo - extend to all transactions
-    }
 
-  start_lsa = prior_lsa_next_record (thread_p, node, tdes);
-  if (VACUUM_IS_THREAD_VACUUM (thread_p))
-    {
-      /* Cache postpone log record. An entry for this postpone log record was already created and we also need to save
-       * its LSA. */
-      vacuum_cache_log_postpone_lsa (thread_p, &start_lsa);
-      // todo - extend to all transactions
-    }
+  /* Cache postpone log record. Redo data must be saved before calling prior_lsa_next_record, which may free this
+   * prior node. */
+  tdes->m_log_postpone_cache.redo_data (node->data_header, node->rdata, node->rlength);
+
+  LOG_LSA start_lsa = prior_lsa_next_record (thread_p, node, tdes);
+
+  /* Cache postpone log record. An entry for this postpone log record was already created and we also need to save
+   * its LSA. */
+  tdes->m_log_postpone_cache.insert (start_lsa);
```

Comment:
I think redo_data and insert LSA can be merged into a single function that is called here.

Reply:
fixed